ddtrace/contrib/botocore/services/bedrock.py

```diff
@@ -81,44 +85,22 @@ def __iter__(self):
             for line in self.__wrapped__:
                 self._body.append(json.loads(line["chunk"]["bytes"]))
                 yield line
-            metadata = _extract_streamed_response_metadata(self._datadog_span, self._body)
-            formatted_response = _extract_streamed_response(self._datadog_span, self._body)
-            self._process_response(formatted_response, metadata=metadata)
-            self._datadog_span.finish()
+            metadata = _extract_streamed_response_metadata(self._execution_ctx, self._body)
+            formatted_response = _extract_streamed_response(self._execution_ctx, self._body)
+            core.dispatch(
+                "botocore.bedrock.process_response",
+                [
+                    self._execution_ctx,
+                    formatted_response,
+                    metadata,
+                    self._body,
```

Comment:
We're passing in `self._body` which is a list of dictionaries so I'm worried this will break `_on_botocore_bedrock_process_response()` which expects a dictionary.

Reply:
Just an FYI, I double checked the bedrock API and we only care about tagging choice ID (only available when) the provider is COHERE and it is not streamed, so we should be fine
i.e. this line should always evaluate to false since we're in the `__iter__` stream handler.